test/language/class/class-strict-mode.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class strict mode
+---*/
+class C {}
+
+with ({a: 1}) {
+  assert.sameValue(a, 1);
+}
+
+assert.throws(SyntaxError,
+             'class C extends function B() { with ({}); return B; }() {}');
```

Comment:
This is probably why Test262 defines [the `negative` flag](https://github.com/tc39/test262/blob/master/CONTRIBUTING.md#negative): so you can write expectations for SyntaxErrors without relying on `eval` (and assuming its correct functioning)